File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/DataStruct/IndexList/IndexItemBase.cpp
#include "MyProject.h"
#include "IndexItemBase.h"

MY_BEGIN_NAMESPACE(MyNS)

IndexItemBase::IndexItemBase()
{
	this->mIndex = -1;
}

void IndexItemBase::init()
{

}

void IndexItemBase::dispose()
{

}

int IndexItemBase::getIndex()
{
	return this->mIndex;
}

void IndexItemBase::setIndex(int value)
{
	this->mIndex = value;
}

void IndexItemBase::resetIndex()
{
	this->mIndex = -1;
}
	
MY_END_NAMESPACE

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/DataStruct/IndexList/IndexItemBase.cs
namespace SDK.Lib
{
    public class IndexItemBase : GObject
    {
        protected int mIndex;   // 索引

        public IndexItemBase()
        {
            this.mIndex = -1;
        }

        override public void init()
        {

        }

        override public void dispose()
        {

        }

        public int getIndex()
        {
            return this.mIndex;
        }

        public void setIndex(int value)
        {
            this.mIndex = value;
        }

        public void resetIndex()
        {
            this.mIndex = -1;
        }
    }
}